facilities/models.py

```diff
@@ -0,0 +1,58 @@
+from django.db import models
+from common.models import AbstractBase, SubLocation
+
+FACILLITY_STATUS = (
+    ('OPERATIONAL', 'Operations are running normally'),
+    ('NOT_OPERATIONAL', 'The facility is not operating'),
+)
+
+FACILITY_TYPES = (
+    ('DISPENSARY', 'dispensary'),
+    ('HEALTH_CENTER', 'Health Center'),
+    ('LEVEL_1', 'Level 1 facility'),
+    ('LEVEL_2', 'Level 2 facility'),
+    ('LEVEL_3', 'Level 3 facility'),
+    ('LEVEL_4', 'Level 4 facility'),
+    ('LEVEL_5', 'Level 5 facility'),
+)
+
+
+class Owner(AbstractBase):
+    name = models.CharField(max_length=100, unique=True)
+    description = models.TextField(null=True, blank=True)
+    code = models.CharField(max_length=100)
+
+
+class Service(AbstractBase):
+    name = models.CharField(max_length=255)
+    description = models.TextField(null=True, blank=True)
+    code = models.CharField(max_length=100, null=True, blank=True)
```

Comment:
Let the `code` be non nullable. In the event that a null value is supplied, supply a callable that can be evaluated to create a default `code` by applying a transformation to the `name`